internal/retry/retry.go

```diff
@@ -80,36 +96,47 @@ func (r *OutboundMiddleware) Call(ctx context.Context, request *transport.Reques
 	boff := policy.backoffStrategy.Backoff()
 
 	for i := uint(0); i < policy.retries+1; i++ {
+		r.observer.attempt()
 		timeout, _ := getTimeLeft(ctx, policy.maxRequestTimeout)
 		subCtx, cancel := context.WithTimeout(ctx, timeout)
 		resp, err = out.Call(subCtx, request)
 		cancel() // Clear the new ctx immdediately after the call
 
-		if err == nil || !isRetryable(err) {
+		if err == nil {
+			r.observer.success()
+			return resp, err
+		}
+
+		if !isRetryable(err) {
+			r.observer.unretryableError()
 			return resp, err
 		}
 
 		// Reset the rereader so we can do another request.
 		if resetErr := rereader.Reset(); resetErr != nil {
+			r.observer.middlewareError()
```

Comment:
What does this mean to a user? Does it indicate a bug in YARPC, or a bug in the retry middleware, or a bug in some other piece of middleware?